heareval/evaluation/task_evaluation.py

```
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+"""
+Compute evaluation metrics on a set of predictions for a task.
+
+TODO: This is a start on this file. Still need to add a bunch of metrics for the tasks
+    including AUC, chroma error for NSYnth pitch, and event-based metrics for DCASE.
+    Would it make sense to move all the metric functions over to a file called metrics?
+"""
+
+import json
+from pathlib import Path
+import pickle
+from typing import Dict, List
+
+import numpy as np
+import pandas as pd
+import torch
+
+
+def top1_error(
+    predictions: np.ndarray, targets: List, label_vocab: pd.DataFrame
+) -> Dict[str, float]:
+
+    # Dictionary of labels and integer idx: {label -> idx}
+    label_vocab = label_vocab.set_index("label").to_dict()["idx"]
+
+    # Compute the number of correct predictions
+    correct = 0
+    for i, prediction in enumerate(predictions):
+        predicted_class = np.argmax(prediction)
+        assert len(targets[i]) == 1
+        target_class = label_vocab[targets[i][0]]
+
+        if predicted_class == target_class:
+            correct += 1
+
+    top1_error = correct / len(targets)
+    return {"top1_error": top1_error}
+
+
+def auc(
+    predictions: np.ndarray, targets: List, label_vocab: pd.DataFrame
+) -> Dict[str, float]:
+    # TODO Compute AUC
+    return {"auc": 0.0}
+
+
+# TODO: Add additional metrics
+
+available_metrics = {"top1_error": top1_error, "auc": auc}
+
+
+def task_evaluation(task_path: Path):
+
+    metadata = json.load(task_path.joinpath("task_metadata.json").open())
+    label_vocab = pd.read_csv(task_path.joinpath("labelvocabulary.csv"))
+
+    embedding_type = metadata["embedding_type"]
+
+    if "evaluation" not in metadata:
+        print(f"Task {task_path.name} has no evaluation config.")
```

This should raise an error